Make supervisor_export_image test image content.

brettle opened the pull request:
Use exported image as a texture and do a basic check for correctness.

I'm also experiencing some strange behavior with image export on my local machine and this is a way to figure out whether it also happens during CI.

stefaniapedrazzi reviewed (commented):
It is good to also test the content of the exported image.
Camera image colors might slightly differ due to the differences in the graphics card.
The test is still failing because of the move handles that are displayed for the selected object.
The selection should be disabled from the supervisor by selecting for example the Viewpoint or the WorldInfo node or it could be set in the WBPROJ file (that in this particular case will have to be committed to the repo).
In the WBPROJ file you simply have to add the line:
```
userInteractions: selectionDisabled
```

Meanwhile, I have a couple of comments about the coding style.

brettle commented:
Thanks for the feedback.

Why would color differ by graphics card in this case? For the image I'm exporting, I'm using a material that emits the pure color and for the image grabbed by the camera, I'm using such a bright light (with no attenuation) that the image should be saturated. If it's not, I can increase the light intensity until it is.

brettle commented:
Any idea why this is still failing on ubuntu-20.04? The exported image which is supposed be green is actually the color of the sky. I added additional planes to completely enclose the viewpoint (in colors that aren't the blue of the sky) and I changed the workflow to make the `tests/` directory available as an artifact if the test suite fails. The `GREEN.png` file from that artifact should be green but is this instead:

![GREEN](https://user-images.githubusercontent.com/118192/218879207-9e9769f4-2d74-4730-abf2-8f6fd2626067.png)

omichel commented on tests/with_rendering/controllers/supervisor_export_image/supervisor_export_image.c:
```diff
@@ -0,0 +1,105 @@
+#include <stdio.h>
+#include <unistd.h>
+
+#define FILE_NAME_PNG "image.png"
+#define FILE_NAME_JPG "image.jpg"
+
```
This doesn't seem to be used.
```suggestion
```

omichel commented on tests/with_rendering/controllers/supervisor_export_image/supervisor_export_image.c:
```diff
@@ -0,0 +1,105 @@
+#define FILE_NAME_PNG "image.png"
+#define FILE_NAME_JPG "image.jpg"
+
+bool file_exists(const char *filename) {
```
```suggestion
static bool file_exists(const char *filename) {
```

omichel commented on tests/with_rendering/controllers/supervisor_export_image/supervisor_export_image.c:
```diff
@@ -0,0 +1,105 @@
+
+  // Turn to face the requested direction
+  WbFieldRef rotation_field = wb_supervisor_node_get_field(wb_supervisor_node_get_from_def("Test"), "rotation");
+  double rotation[4] = {0.0, 0.0, 1.0, 0.0};
+  rotation[3] = angle_radians;
```
```suggestion
  double rotation[4] = {0.0, 0.0, 1.0, angle_radians};
```

omichel commented on tests/with_rendering/worlds/supervisor_export_image.wbt:
```diff
@@ -0,0 +1,92 @@
+  ambientIntensity 1
+  intensity 10
+}
+DEF Test Robot {
```
We generally use all caps for DEF names.
```suggestion
DEF TEST Robot {
```

omichel commented on tests/with_rendering/controllers/supervisor_export_image/supervisor_export_image.c:
```diff
@@ -0,0 +1,105 @@
+  ts_assert_boolean_not_equal(file_exists(file_name_jpg), "Impossible to remove the JPG file");
+
+  // Turn to face the requested direction
+  WbFieldRef rotation_field = wb_supervisor_node_get_field(wb_supervisor_node_get_from_def("Test"), "rotation");
```
```suggestion
  WbFieldRef rotation_field = wb_supervisor_node_get_field(wb_supervisor_node_get_from_def("TEST"), "rotation");
```

omichel reviewed (approved):
Thank you.
